sootup.core/src/main/java/sootup/core/jimple/visitor/ReplaceUseStmtVisitor.java

```diff
@@ -96,10 +96,14 @@ public void caseAssignStmt(@Nonnull JAssignStmt stmt) {
       if (rValue == oldUse) {
         setResult(stmt.withRValue(newUse));
       } else {
-        refVisitor.init(oldUse, newUse);
-        ((Ref) rValue).accept(refVisitor);
-        if (refVisitor.getResult() != rValue) {
-          setResult(stmt.withRValue(refVisitor.getResult()));
+        try {
+          refVisitor.init(oldUse, newUse);
+          ((Ref) rValue).accept(refVisitor);
+          if (refVisitor.getResult() != rValue) {
+            setResult(stmt.withRValue(refVisitor.getResult()));
+          }
+        } catch (ClassCastException cce) {
+          // can not replace that local by another Value
```

Comment:
empty catch block

Reply:
log a warning?

Reply:
we don't have to log as we use this try catch as a means to "is it possible to apply this new value" and if it fails it sets no new result, which conforms to the other uses there